text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,545 @@
+to the service team to get their input and plans for the service. 
+
+It’s a large effort (X devs * 1 week) to invest in a module when we have other pressing
+projects. However, the bulk of the effort has been spent already since we have fairly
+robust prototypes already implemented.
+
+### What is the technical solution (design) of this feature?
+
+
+```mermaid
+classDiagram
+    Expression <|-- Schedule : scheduleExpression
+    Key <|-- Schedule : key
+    ScheduleTargetBase <|-- Schedule : target
+    Group <|-- Schedule : group
+    Input <|-- ScheduleTargetBase : input
+    Targets <.. ScheduleTargetBase : implements
+
+    class Schedule {
+        + String name
+        + Duration flexibleTimeWindow 
+        + Date startDate
+        + Date endDate
+        + String scheduleTimeZone 
+        + Boolean disabled 
+
+        + <> targetOverrideProps
+
+        +grant(..)
+        +grantRead(..)
+        +grantWrite(..)
+        +Metric metric..() 
+    }
+    class Expression {
+        + Expression at(date: Date)
+        + Expression rate(duration: Duration) 
+        + Expression cron(..)
+        + Expression expression(..)
+    }
+
+    class Group {
+        + String name
+        + addSchedules(..Schedule)
+
+        + Metric metric..()
+        +grant(..)
+        +grantRead(..)
+        +grantWrite(..)
+    }
+
+    class ScheduleTargetBase {
+        + sqs.IQueue deadLetterQueue
+        + Duration maximumEventAge
+        + Number maximumRetryAttempts
+    }
+
+    class Input {
+        + Input fromText(..)
+        + Input fromMultilineText(..)
+        + Input fromObject(..)
+        + Input fromInput(..)
+    }
+```
+
+![L2 Constructs Class Diagram](../images/EventBridge-Scheduler-2023-03-05-1723.excalidraw.png)
+
+Prototype at https://github.com/filletofish/cdk-eb-scheduler/
+
+
+### Is this a breaking change?
+
+No.
+
+### What alternative solutions did you consider?
+
+> Briefly describe alternative approaches that you considered. If there are
+> hairy details, include them in an appendix.
+
+1. We have discussed an opportunity of reusing existing `events.targets` module: https://docs.aws.amazon.com/cdk/api/v1/docs/aws-events-targets-readme.html. See RFC comments
+
+### What are the drawbacks of this solution?
+
+> Describe any problems/risks that can be introduced if we implement this RFC.
+
+TBD
+
+### What is the high-level project plan?
+
+> Describe your plan on how to deliver this feature from prototyping to GA.
+> Especially think about how to "bake" it in the open and get constant feedback
+> from users before you stabilize the APIs.
+>
+> If you have a project board with your implementation plan, this is a good
+> place to link to it.
+
+Most part of the L2 constructs is already implemented. We have a set of relatively (1-2 day of work) todos that can be implemented in parallel: 
+
+1. Decide what removal policy for Group should be 
```

Comment:
Do we need to discuss this any further?

Reply:
We need to do some tests. Main point to figure out is if a group you create will actually delete all schedules in it when the group is deleted. That would make it a resource with state and its state would deserve protection.